app/src/main/java/com/example/taskthree/view/GuessFragment.kt

```diff
@@ -0,0 +1,109 @@
+package com.example.taskthree.view
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.activityViewModels
+import androidx.navigation.fragment.findNavController
+import com.example.taskthree.databinding.FragmentCharBinding
+import com.example.taskthree.viewmodel.GuessViewModel
+
+class GuessFragment : Fragment() {
+    private lateinit var binding: FragmentCharBinding
+    private val viewModel: GuessViewModel by activityViewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        binding = FragmentCharBinding.inflate(inflater, container, false)
+        buttonConnectEditText()
+
+        viewModel.resultMessageObs.observe(viewLifecycleOwner) {
+            if (it == TRY_AGAIN_MESSAGE) {
+                binding.textViewGuessHint.text = viewModel.randomCharObs.toString()
+            } else if (it == WIN_MESSAGE){
+                binding.textViewGuessHint.text=
+                    StringBuilder().append(GUESS_CHARACTER_HINT+ viewModel.randomIntObs.value.toString())
```

Comment:
if (it == TRY_AGAIN_MESSAGE)  gibi içerisinde "it" olan fonksiyonların daha açıklayıcı olması açısından değiştirlmesi önerilir. Misal bu örnekte it dediğimiz değişken aslında bir mesaj.